server/src/api_endpoints/clients/roles.js

```diff
@@ -13,11 +10,21 @@ module.exports = {
  * @param {*} res
  * @param {*} next
  */
-async function listClientRoles (req, res, next) {
-  const clientId = req.params.client_id
-
-  const roles = await dbAdminRole.listRoles(clientId)
-  res.status(200).send({ data: roles })
+const listClientRoles = async (req, res, next) => {
+  try {
+    const { client_id: clientId } = req.params
+    const { include_permissions } = req.query
+    await checkClientExists(clientId)
+    const roles = await dbAdminRoles.listRoles(clientId)
+    if (include_permissions === 'true') {
+      for (const role of roles) {
+        role.permissions = await dbAdminPermissions.getPermissionsByRoleKey(role.key)
+      }
+    }
+    res.status(200).send({ data: roles })
```

Comment:
@mikelax We are not using this API as far as I know. However, since this is a public project and fixing these inconsistencies would be breaking changes, I think we should consider addressing them when we decide to have a major release